tests/test_criterion_result.ts

```diff
@@ -0,0 +1,250 @@
+    criterion = await Criterion.create(handgrading_rubric.pk, {});
+
+    observer = new TestObserver();
+    CriterionResult.subscribe(observer);
+});
+
+afterEach(() => {
+    CriterionResult.unsubscribe(observer);
+});
+
+describe('List/create criterion result tests', () => {
+    test('Criterion result ctor', () => {
+        let now = (new Date()).toISOString();
+        let expected_student_file = new CriterionResult({
+            pk: 6,
+            last_modified: now,
+            selected: false,
+            criterion: criterion,
+            handgrading_result: handgrading_result.pk,
+        });
+
+        expect(expected_student_file.pk).toEqual(6);
+        expect(expected_student_file.last_modified).toEqual(now);
+        expect(expected_student_file.selected).toEqual(false);
+        expect(expected_student_file.criterion).toEqual(criterion);
+        expect(expected_student_file.handgrading_result).toEqual(handgrading_result.pk);
+    });
+
+    test('List criterion result', async () => {
+        let create_criterion_result = `
+from autograder.handgrading.models import (
+    HandgradingResult, CriterionResult, Criterion, HandgradingRubric
+)
+rubric = HandgradingRubric.objects.get(pk=${handgrading_rubric.pk})
+result = HandgradingResult.objects.get(pk=${handgrading_result.pk})
+
+c2 = Criterion.objects.validate_and_create(handgrading_rubric=rubric)
+c3 = Criterion.objects.validate_and_create(handgrading_rubric=rubric)
+
+CriterionResult.objects.validate_and_create(handgrading_result=result, selected=True, criterion=c2)
+CriterionResult.objects.validate_and_create(handgrading_result=result, selected=True, criterion=c3)
+        `;
+        run_in_django_shell(create_criterion_result);
+
+        let loaded_criterion_results = await CriterionResult.get_all_from_handgrading_result(
+            handgrading_result.pk);
+
+        let sorted_criterion_results = loaded_criterion_results.sort((a, b) => a.pk - b.pk);
+        expect(sorted_criterion_results.length).toEqual(3);
+        expect(sorted_criterion_results[0].pk).toEqual(1);
+        expect(sorted_criterion_results[0].selected).toEqual(false);
+        expect(sorted_criterion_results[0].handgrading_result).toEqual(handgrading_result.pk);
+        expect(sorted_criterion_results[0].criterion).toEqual(criterion);
+
+        expect(sorted_criterion_results[1].pk).toEqual(2);
+        expect(sorted_criterion_results[1].selected).toEqual(true);
+        expect(sorted_criterion_results[1].handgrading_result).toEqual(handgrading_result.pk);
+
+        expect(sorted_criterion_results[2].pk).toEqual(3);
+        expect(sorted_criterion_results[2].selected).toEqual(true);
+        expect(sorted_criterion_results[2].handgrading_result).toEqual(handgrading_result.pk);
+    });
+});
+
+describe('Get/update/delete criterion result tests', () => {
+    let criterion_result!: CriterionResult;
+
+    /** NOTE: This assumes pks are assigned sequentially, and so CriterionResult with pk=1
```

Comment:
A less fragile way to get around the lack of a create function would be to make the python code that creates the criterion result print the pk of the object (and nothing else). Then you can use `.stdout` to access the output of `run_in_django_shell` and parse it as an int.